### PR TITLE
[DOCS] Precision Control article for master

### PR DESCRIPTION
**Pull request by msmykx-intel**
Ticket: 101682

* Added separate article on Precision Control (ov::hint::execution_mode and ov::inference_precision properties)

### CODE REVIEW

File: docs/OV_Runtime_UG/precision_control.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Separately, you can control the IR precision. Essentially the IR precision becomes a way to compress your model by reducing the precision of the weights, and it doesn't affect how devices execute the model.
+
+This change clears up a lot of confusion where, for example, you couldn't execute a high-performance model on the GPU by default, and the behavior between devicess was different.
+
+.. note::
+
+   All devices (except GNA) only support floating-point data types (``f32``, ``f16``, ``bf16``) as a value for ``inference_precision`` attribute, because quantization cannot be done in Runtime. The GNA plugin has the ability to perform model quantization on ``core.compile_model()`` call, so it supports integer data types in addition to ``f32``.
```
Comment (vladimir-paramuzov) [inline docs/OV_Runtime_UG/precision_control.md:13]: I think this should be a part of "Inference Precision" paragraph

File: docs/OV_Runtime_UG/precision_control.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Execution Mode
+##############
+
+``ov::hint::execution_mode`` is a high-level hint to control whether the user wants to keep the best accuracy (**ACCURACY mode**) or if the plugin can do some optimizations that may lower the accuracy for performance reasons (**PERFORMANCE mode**)
```
Comment (vladimir-paramuzov) [inline docs/OV_Runtime_UG/precision_control.md:18]: Let's add more details.
In ACCURACY mode plugin is not allowed to do conversions of floating point tensors to smaller floating point type, so the plugins tries to keep accuracy metrics as close as possible to the original values received after training with respect to the actual device capabilities. That means that most of the plugins will do inference  in fp32 precision if the device supports it. 
In PERFORMANCE mode plugin may do conversion to smaller data types and apply some other optimizations which may have certain impact on the accuracy metrics, though we still try to minimize accuracy drop and in some cases may use mixed precision execution.

File: docs/OV_Runtime_UG/precision_control.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Precision Control {#openvino_docs_OV_UG_Precision_Control}
+
+@sphinxdirective
+
+Regardless of IR precision, devices will run in high performance mode by default. For GPU this means ``fp16`` inference and for CPU - ``bf16`` inference (if available). Previously, you had to convert the IR to ``fp16`` for the GPU to run in ``fp16``, and for the CPU, it was high precision by default. Now the devices have been aligned and this selection has been disconnected from IR precision. If high performance is causing the accuracy issue (only seen a few times in history), you can use the ``inference_precision`` hint and set it to accuracy.
+
+Separately, you can control the IR precision. Essentially the IR precision becomes a way to compress your model by reducing the precision of the weights, and it doesn't affect how devices execute the model.
+
+This change clears up a lot of confusion where, for example, you couldn't execute a high-performance model on the GPU by default, and the behavior between devicess was different.
+
```
Comment (vladimir-paramuzov) [inline docs/OV_Runtime_UG/precision_control.md:10]: I'd add few words on why usage of lower precision is important for the performance: compute bandwidth is usually higher with smaller data types and hw often have special blocks for efficient multiply-accumulate operations with smaller data types only (e.g. XMX on GPU and AMX on CPU don't support fp32) + less memory I/O operations due to smaller tensor byte size.

File: docs/OV_Runtime_UG/precision_control.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Precision Control {#openvino_docs_OV_UG_Precision_Control}
+
+@sphinxdirective
+
+Regardless of IR precision, devices will run in high performance mode by default. For GPU this means ``fp16`` inference and for CPU - ``bf16`` inference (if available). Previously, you had to convert the IR to ``fp16`` for the GPU to run in ``fp16``, and for the CPU, it was high precision by default. Now the devices have been aligned and this selection has been disconnected from IR precision. If high performance is causing the accuracy issue (only seen a few times in history), you can use the ``inference_precision`` hint and set it to accuracy.
```
[the remark this answers is not in the document]
Comment (msmykx-intel) [reply docs/OV_Runtime_UG/precision_control.md]: Done in https://github.com/openvinotoolkit/openvino/pull/17413/commits/42249732907664d595498e427f8288c168c70e4c

File: docs/OV_Runtime_UG/precision_control.md (hunk 2 above)
Comment (msmykx-intel) [reply docs/OV_Runtime_UG/precision_control.md]: Done in https://github.com/openvinotoolkit/openvino/pull/17413/commits/42249732907664d595498e427f8288c168c70e4c

File: docs/OV_Runtime_UG/precision_control.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+This change clears up a lot of confusion where, for example, you couldn't execute a high-performance model on the GPU by default, and the behavior between devicess was different.
+
+Execution Mode
+##############
+
+``ov::hint::execution_mode`` is a high-level hint to control whether the user wants to keep the best accuracy (**ACCURACY mode**) or if the device can do some optimizations that may lower the accuracy for performance reasons (**PERFORMANCE mode**)
+
+* In **ACCURACY mode**, the device cannot convert floating point tensors to a smaller floating point type, so devices try to keep the accuracy metrics as close as possible to the original values ​​obtained after training relative to the device's real capabilities. This means that most devices will infer with ``fp32`` precision if your device supports it.
+* In **PERFORMANCE mode**, the device can convert to smaller data types and apply other optimizations that may have some impact on accuracy rates, although we still try to minimize accuracy loss and may use mixed precision execution in some cases.
```
Comment (vladimir-paramuzov) [inline docs/OV_Runtime_UG/precision_control.md:24]: Let's add a few words about quantization:

```
If a model was quantized using OpenVINO optimization tools (ref to https://docs.openvino.ai/latest/ptq_introduction.html) or any other method, then quantized operators will be executed in the target integer precision if the device has HW acceleration for that type. 
For example, quantized int8 primitives are executed in int8 precision for both ACCURACY and PERFORMANCE modes if the device provides higher compute bandwidth for 8-bit data types comparing to any available floating-point type. 
On the other hand, devices with no HW acceleration for int8 data type may keep such operators in floating-point precision, and the exact floating-point type will be impacted by execution_mode and inference_precision properties. 
```

File: docs/OV_Runtime_UG/precision_control.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+Inference Precision
+###################
+
+``ov::hint::inference`` precision is a lower-level property that allows you to specify the exact precision the user wants, but is less portable. For example, CPU supports ``f32`` inference precision and ``bf16`` on some platforms, GPU supports ``fp32`` and ``fp16`` while GNA supports ``i8`` and ``i16``, so if a user wants to an application that uses multiple devices, they have to handle all these combinations manually or let OV do it automatically by using higher level ``execution_mode`` property. Another thing is that ``inference_precision`` is also a hint, so the value provided is not guaranteed to be used by Runtime (mainly in cases where the current device does not have the required hardware capabilities).
```
Comment (vladimir-paramuzov) [inline docs/OV_Runtime_UG/precision_control.md:29]: ``ov::hint::inference_precision``

File: docs/OV_Runtime_UG/precision_control.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+Previously, these 2 precisions were interrelated, and model storage precision could affect the inference precision in some devices (e.g. GPU did ``f16`` inference only for ``f16`` IRs).
+
+With the ``2023.0`` release this behavior has been changed and the inference precision no longer depends on the precision of IR. Now users have several knobs to find the balance between model performance and accuracy.
+
+Essentially, the IR precision becomes a way of compressing the model by reducing the precision of the weights, and it does not affect how the devices execute the model. And using lower precision is important for performance because compute bandwidth tends to be higher for smaller data types, and hardware often has special blocks for efficient multiply-accumulate operations with smaller data types only (e.g. XMX on GPU and AMX on CPU do not support ``fp32``). Also, I/O operations requires less memory due to the smaller tensor byte size. This guide will focus on how to control inference precision.
+
+This change clears up a lot of confusion where, for example, you couldn't execute a high-performance model on the GPU by default, and the behavior between devicess was different.
```
[the remark this answers is not in the document]
Comment (msmykx-intel) [reply docs/OV_Runtime_UG/precision_control.md]: Done in https://github.com/openvinotoolkit/openvino/pull/17413/commits/16a5f93b77fc0bb4202b127846568552387fd060

File: docs/OV_Runtime_UG/precision_control.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+1. Model storage precision (IR precision),
+2. Model inference precision.
+
+Previously, these 2 precisions were interrelated, and model storage precision could affect the inference precision in some devices (e.g. GPU did ``f16`` inference only for ``f16`` IRs).
+
+With the ``2023.0`` release this behavior has been changed and the inference precision no longer depends on the precision of IR. Now users have several knobs to find the balance between model performance and accuracy.
+
+Essentially, the IR precision becomes a way of compressing the model by reducing the precision of the weights, and it does not affect how the devices execute the model. And using lower precision is important for performance because compute bandwidth tends to be higher for smaller data types, and hardware often has special blocks for efficient multiply-accumulate operations with smaller data types only (e.g. XMX on GPU and AMX on CPU do not support ``fp32``). Also, I/O operations requires less memory due to the smaller tensor byte size. This guide will focus on how to control inference precision.
```
[the remark this answers is not in the document]
Comment (msmykx-intel) [reply docs/OV_Runtime_UG/precision_control.md]: Done in https://github.com/openvinotoolkit/openvino/pull/17413/commits/16a5f93b77fc0bb4202b127846568552387fd060

File: docs/OV_Runtime_UG/precision_control.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (msmykx-intel) [reply docs/OV_Runtime_UG/precision_control.md]: Done in https://github.com/openvinotoolkit/openvino/pull/17413/commits/16a5f93b77fc0bb4202b127846568552387fd060

File: docs/OV_Runtime_UG/precision_control.md (hunk 5 above)
Comment (msmykx-intel) [reply docs/OV_Runtime_UG/precision_control.md:25]: Done in https://github.com/openvinotoolkit/openvino/pull/17413/commits/16a5f93b77fc0bb4202b127846568552387fd060

File: docs/OV_Runtime_UG/precision_control.md (hunk 6 above)
Comment (msmykx-intel) [reply docs/OV_Runtime_UG/precision_control.md]: Done in https://github.com/openvinotoolkit/openvino/pull/17413/commits/16a5f93b77fc0bb4202b127846568552387fd060

**Review by vladimir-paramuzov — APPROVED** [review]

**Review by kblaszczak-intel — APPROVED** [review]

**Pull request merged**